ckan/cli/generate.py

```diff
@@ -2,35 +2,51 @@
 
 from __future__ import print_function
 import os
-import sys
+
+import alembic.command
 import click
+from alembic.config import Config as AlembicConfig
+
+import ckan
+from ckan.cli.db import _resolve_alembic_config
+import ckan.plugins.toolkit as tk
+
 import uuid
 import string
 import secrets
 from ckan.cli import error_shout
 
 
-@click.group(
-    name=u'generate',
-    short_help=u"Generate empty extension files to expand CKAN.",
-    invoke_without_command=True,
-)
+class CKANAlembicConfig(AlembicConfig):
+    def get_template_directory(self):
+        return os.path.join(os.path.dirname(ckan.__file__),
+                            u"../contrib/alembic")
+
+
+@click.group()
 def generate():
-    try:
-        from cookiecutter.main import cookiecutter
-    except ImportError:
-        error_shout(u"`cookiecutter` library is missing from import path.")
-        error_shout(u"Make sure you have dev-dependencies installed:")
-        error_shout(u"\tpip install -r dev-requirements.txt")
-        raise click.Abort()
+    """Scaffolding for regular development tasks.
+    """
+    pass
 
 
 @generate.command(name=u'extension', short_help=u"Create empty extension.")
-@click.option(u'-o', u'--output-dir', help=u"Location to put the generated "
-                                           u"template.",
+@click.option(u'-o',
+              u'--output-dir',
+              help=u"Location to put the generated "
+              u"template.",
               default=u'.')
 def extension(output_dir):
-    from cookiecutter.main import cookiecutter
+    """Generate empty extension files to expand CKAN.
+    """
+    try:
+        from cookiecutter.main import cookiecutter
+    except ImportError:
+        tk.error_shout(u"`cookiecutter` library is missing from import path.")
+        tk.error_shout(u"Make sure you have dev-dependencies installed:")
+        tk.error_shout(u"\tpip install -r dev-requirements.txt")
+        raise click.Abort()
```

Comment:
Ah, this has been moved out of the decorator here as well as in my PR #5381 @kowh-ai 👍